src/index.ts

```diff
@@ -1,3 +1,61 @@
-export default function greeter(name: string): string {
-  return `Hello, ${name}!`;
-}
+import openrpcDocument from './openrpc.json';
+import methodMapping from './methods/methodMapping';
+
+window.addEventListener('message', async (event: MessageEvent) => {
+  if (!event.data.jsonrpc) {
+    return;
+  }
+
+  const eventSource: Window = event.source as Window;
+
+  if (event.data.method === 'rpc.discover') {
+    eventSource.postMessage(
+      {
+        jsonrpc: '2.0',
+        result: openrpcDocument,
+        id: event.data.id,
+      },
+      event.origin,
+    );
+    return;
+  }
+
+  if (!methodMapping[event.data.method]) {
+    eventSource.postMessage(
+      {
+        jsonrpc: '2.0',
+        error: {
+          code: 32009,
```

Comment:
It looks like [JSON-RPC](https://www.jsonrpc.org/specification#response_object) uses the code -32601 for "Method not found" :thinking:

Reply:
Also curious where the code `32329` comes from, which is used below

Reply:
not sure why i used these codes, i used this as the starting point from back when I did the snaps hackathons: https://github.com/xops/inspector-snaps-transport/blob/master/src/index.tsx#L19

we should probably use the same jsonrpc eth errors lib we used for metaRPCHandler:

https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/createMetaRPCHandler.js#L11-L13

Reply:
maybe the generic ones like method not found can be pulled into another library?

Reply:
I guess technically its still ethereum based codes but we are re-using them as convention